tools/prow-config-updater/pullrequest.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/google/go-github/github"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/pkg/test/cmd"
+	"knative.dev/pkg/test/ghutil"
+	"knative.dev/pkg/test/helpers"
+
+	"knative.dev/test-infra/shared/git"
+	"knative.dev/test-infra/tools/prow-config-updater/config"
+)
+
+type GitHubMainHandler struct {
+	client *ghutil.GithubClient
+	info   git.Info
+	dryrun bool
+}
+
+// Get the latest pull request created in this repository.
+// TODO(chizhg): get rid of this hack once Prow supports setting PR number as an env var for postsubmit jobs.
+func (gc *GitHubMainHandler) getLatestPullRequest() (*github.PullRequest, error) {
```

Comment:
This is not guaranteed to be accurate, I think we need something better

Reply:
Yes, that's why I have that TODO comment, but this will work for now.

Reply:
that's ok. Would expect a comment explaining that it might be wrong

Reply:
Added.